juju/juju
# model manager facade service

Introduces model service to model facade.
    
With this commit we are introducing the model service into the model
manager facade and starting the process of developing the full model
creation process in Dqlite.
    
The way this PR is implemented is temporary by where we have separated
the DQlite logic from the mongo logic in the facade. This is so that we
can remove the modelmanager service from the facade and later implement
the model DDL changes needed.
    
In a follow up PR we will move the new DQlite logic back into the main
facade call and remove mongo.
    
This commit also takes away the control from callers of create model
specifying what type a cloud will be and instead leaves this logic up to
the service to decide. This is important business logic that should be
being made in the service.

## Checklist


## QA steps

We are changing the services used for facade calls when creating a new model. To test this PR we want to bootstrap a new controller and confirm we can successfully add a new model.

## Documentation changes

N/A

## Links


**Jira card:** JUJU-5341
